tests/encodeinput.cpp

```diff
@@ -167,40 +167,47 @@ EncodeOutput::~EncodeOutput()
         fclose(m_fp);
 }
 
-EncodeOutput* EncodeOutput::create(const char* outputFileName, int width , int height)
+EncodeOutput* EncodeOutput::create(const char* outputFileName, int width,
+                                   int height, const char* codecName)
 {
-    EncodeOutput * output = NULL;
-    if(outputFileName==NULL)
+    EncodeOutput* output = NULL;
+    if (outputFileName == NULL)
         return NULL;
-    const char *ext = strrchr(outputFileName,'.');
-    if(ext==NULL)
+    const char* ext = strrchr(outputFileName, '.');
+    std::string codec(""); // needed for strcasecmp to work
```

Comment:
Actually this is the scenario:

Calling create without a codec name will result in codecName == ""
Calling create with codec but no codec given in command line results on codecName == NULL which requires a auxiliary std::string name to copy the "" for strcmpcase to work
Calling create with codec and codec given in command line results on codecName == "some name" which is ok.

Let me know where you want to handle the extra copy.  I think we can leave it here.  The other option is to add it to encodehelp.h when codec ==NULL